src/stores/api.js

```diff
@@ -1,23 +1,51 @@
-import axios from 'axios'
+import axios from "axios";
 
-const BASE_URL = 'http://localhost:8082'
+const BASE_URL = "http://localhost:8082";
 
 export async function getTasks() {
-    const res = await axios.get(BASE_URL + '/api/tasks', {raw:true})
-    return res.data
+  const res = await axios.get(BASE_URL + "/api/tasks", { raw: true });
+  return res.data;
 }
 
 export async function getLatestContributions() {
-    const res = await axios.get(BASE_URL + '/api/contributions/?limit=5', {raw:true})
-    return res.data
+  const res = await axios.get(BASE_URL + "/api/contributions/?limit=5", {
+    raw: true,
+  });
+  return res.data;
 }
 
 export async function getMyContributions(userName) {
-    const res = await axios.get(BASE_URL + '/api/contributions/' + userName, {raw:true})
-    return res.data
+  const res = await axios.get(BASE_URL + "/api/contributions/" + userName, {
+    raw: true,
+  });
+  return res.data;
 }
 
 export async function getAllTimeGreat() {
-    const res = await axios.get(BASE_URL + '/api/contributions/top-scores', {raw:true})
-    return res.data
-}
\ No newline at end of file
+  const res = await axios.get(BASE_URL + "/api/contributions/top-scores", {
+    raw: true,
+  });
+  return res.data;
+}
+
+export async function getLoggedInUser() {
+  const res = await axios
+    .get(BASE_URL + "/api/user", { raw: true })
+    .catch(function (error) {
+      if (error.response.status == 401) {
+        localStorage.setItem("currentUser", null);
+      }
```

Comment:
Hey, @HWaruguru, I am only just starting to dig into the code, but my first question is why you set the currentUser to be null if no user is currently logged in?

What I'm noticing is that if I then reload the page, the "login" button will be replaced by the circular button and I'll get a drop-down menu that shows the username ("null" in this case) and a "logout" option.

If we need to have a null user, you might want to arrange things so that the "login" button only changes if "currentUser" is something other than null.

Reply:
localstorage only stores values as strings, so this is giving currentUser a string value ("null") and not NULL.  That's why this isn't working as you expected.  (Now I've looked at Navbar.vue and see what's supposed to be happening as far as the button rendering goes.)

Reply:
> Hey, @HWaruguru, I am only just starting to dig into the code, but my first question is why you set the currentUser to be null if no user is currently logged in?
> 
> What I'm noticing is that if I then reload the page, the "login" button will be replaced by the circular button and I'll get a drop-down menu that shows the username ("null" in this case) and a "logout" option.
> 
> If we need to have a null user, you might want to arrange things so that the "login" button only changes if "currentUser" is something other than null.

Hey, @nbarnabee thank you for this. I was using it to debug and I forgot to remove it.